### ISSUE TITLE
Difficulties pre-training with filter bank features

### ISSUE THREAD

**Issue by apoorv2904** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#30]
Hi,

First of all thank you for a very clean code base for your work.
I am trying to pre-train a mode with filter-bank  features for librispeech using the provided config upstream_fbankBase.yaml.
The only change I have made is setting
```
train_set:  ['train-clean-100', 'train-clean-360', 'train-other-500']
```

However, my loss seems to be stuck around 0.98. I was wondering if you could tell me the expected loss on filterbank features. 
The pre-trained model also seems to doing bad on montreal-phone classification tasks.

Note that, I was able to pretrain using mel-160 features for same dataset and downstream task.

Thank you for your help on this.

Best Regards,
Apoorv

**Comment by apoorv2904** [comment]
A few more details:
--  I had extracted features using Kaldi and set them up using the instructions provided.
--  I changed gelu to torch.nn.functional.gelu
--  using torch.nn.LayerNorm as TransformerLayerNorm

Since, I could get the mel-160 features to work with the above changes, I don't think this should be the reason why filterbank features are not working.

--Apoorv

**Comment by andi611** [comment]
Hi,

The value of loss depends on the feature, hence loss computed from fbank is different then loss computed from mel-160.

The reason why you did bad on the Montreal-phone set is the following:
The fbank features are pre-processed with a stride of 10 (a.k.a a fbank vector for every 10 ms, a default of Kaldi), which does not match the phone alignment labels of the Montreal-phone set.

On the other hand, the mel-160 feature matches the duration of the Montreal-phone set, that's why you obtained reasonable results when pretrain using mel-160 features.

You should either evaluate with the cpc phone task (suggested approach), or change the parameters in [preprocess_alignment.py](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/b72a2cd9fa106d77f4011c75432eecacc927552c/preprocess/preprocess_alignment.py#L95-L109) to generate a phone label for every 10 ms (a more complicated approach).

I hope this helps!

**Comment by apoorv2904** [comment]
Hi,

Thank you for the clarification. After changing to cpc-phone set,  I get the mentioned classification accuracy. 

I will close the issue.

Thanks,
Apoorv
